src/mongo/tree/MongoAccountTreeItem.ts

```diff
@@ -78,6 +79,20 @@ export class MongoAccountTreeItem implements IAzureParentTreeItem {
 
         throw new UserCancelledError();
     }
+
+}
+
+function validateDatabaseName(database: string): string | undefined | null {
+    // https://docs.mongodb.com/manual/reference/limits/#naming-restrictions
+    const min = 1;
+    const max = 64;
+    if(!database || database.length < min || database.length > max) {
+        return `The name must be between ${min} and ${max} characters.`;
+    }
+    if (/[/\\. "$*<>:|?]/.test(database)) {
```

Comment:
Note : We allow the user to attach to their own mongo servers too. This can reside on another cloud, on-prem, independent of Azure. 

I understand you are implementing the more stringent Windows restrictions. Given that these are placeholder validations, we feel the less restrictive ones (Linux) would be better.  We feel denying a valid resource name could hinder the experience more than allowing a bad use case (an error is thrown).